#summary How to assess transposable element polymorphism in a population

<wiki:toc max_depth="2" />

= Introduction =

A simple and cheap approach to genome-wide assess TE insertion frequencies in populations will allow addressing fundamental questions about TE dynamics and species evolution in more general.

Here we introduce a novel and cost efficient approach to estimate TE population frequencies for TE insertions that are present in the reference sequence as well as for novel TE insertions. Our approach solely requires sequencing of a single paired-end (PE) sample of a pooled population per investigated population. Our approach has the additional advantage that sequencing of pooled populations also yields genome-wide estimates of SNP frequencies and thus standard population genetics measures such as Tajimaâ€™s D can easily be calculated [http://code.google.com/p/popoolation/]. 

= Requirements =

  * paired-end sequences of a pooled population
  * a reference genome
  * a transposable element sequence database (i.e.: a fasta file containing sequences of TEs)
  * Perl
  * BWA
  * !SAMtools
  * !RepeatMasker
  * !PoPoolation TE


= Workflow =

== Preparing the reference sequence ==

  # Download a reference genome. Remove unnecessary comments from the fasta headers
{{{
cat dmel-all-chromosome-r5.31|awk '{print $1}' > ../wg/dmel-5.31-short.fasta
}}}
  # Download the transposon sequences. Remove unnecessary comments from the fasta header
  # Filter TE elements that are to short (e.g.: < 40bp)
  # RepeatMask the reference genome using the transposon sequences as a custom library
{{{
perl RepeatMasker -no_is -nolow -norna --lib te-sequences.fasta -pa 4 reference-genome.fasta
}}}
  # Merge the repeat-masked reference genome and the TE sequences
{{{
cat reference-genome.fasta.masked te-sequences.fasta > combined-reference.fasta
}}}


== Obtaining a TE hierarchy ==

Create a TE hierarchy containing one entry for EVERY sequence in the TE sequences fasta-file! The depth of the hierarchy, the classification of the elements and the names for different hierarchy-levels is up to the user!

example of an hierarchy file:
{{{
insert	id	family	superfamily	suborder	order	class	problem
FBti0015567	Tirant	Tirant	Tirant	LTR	LTR	RNA	0	
FBti0018861	17.6	17.6	17.6	LTR	LTR	RNA	0	
FBti0018862	17.6	17.6	17.6	LTR	LTR	RNA	0	
FBti0018865	297	297	297	LTR	LTR	RNA	0	
FBti0018869	3S18	3S18	3S18	LTR	LTR	RNA	0
FBti0018957	baggins	baggins	baggins	non-LTR	non-LTR	RNA	0
FBti0018944	Tc1	Tc1	Tc1	TIR	TIR	DNA	0
}}}

The first row specifies the names of the hierarchy-levels in ascending order. The problem column may be ignored
All other columns refer to individual TE sequences and their hierarchical status. For every entry in the TE-sequences fasta file the TE-hierarchy has to contain exactly one entry. Note that many individual entries may be present for TE insertions of the same family. The software tries to handle the resulting sequence ambiguity by moving in this TE-hierarchy.

== Mapping the sequences ==

  # Index the reference genome
{{{
bwa index combined-reference.fasta
}}} 
  # Map the PE reads separately using BWA SW (Smith-Waterman algorithm)
{{{
bwa bwasw -t 6 combined-reference.fasta ../reads/Jul_s_5_1_sequence.txt > Jul_s_5_1_seq.sam
bwa bwasw -t 6 combined-reference.fasta ../reads/Jul_s_5_2_sequence.txt > Jul_s_5_2_seq.sam
}}}
  # Create paired-end information using samro (!PoPoolationTE)
{{{
perl samro.pl --sam1 Jul_s_5_1_seq.sam --sam2 Jul_s_5_2_seq.sam --fq1 ../reads/Jul_s_5_1_s
equence.txt --fq2 ../reads/Jul_s_5_2_sequence.txt --output pe-reads.sam
}}}
== Identify TE insertions ==

== Estimate population frequencies ==