#summary How to assess transposable element polymorphism in a population

<wiki:toc max_depth="2" />

= Introduction =

A simple and cheap approach to genome-wide assess TE insertion frequencies in populations will allow addressing fundamental questions about TE dynamics and species evolution in more general.

Here we introduce a novel and cost efficient approach to estimate TE population frequencies for TE insertions that are present in the reference sequence as well as for novel TE insertions. Our approach solely requires sequencing of a single paired-end (PE) sample of a pooled population per investigated population. Our approach has the additional advantage that sequencing of pooled populations also yields genome-wide estimates of SNP frequencies and thus standard population genetics measures such as Tajimaâ€™s D can easily be calculated [http://code.google.com/p/popoolation/]. 

= Requirements =

  * paired-end sequences of a pooled population
  * a reference genome
  * a transposable element sequence database (i.e.: a fasta file containing sequences of TEs)
  * Perl
  * BWA
  * SAMtools
  * RepeatMasker
  * PoPoolation TE


= Workflow =

== Preparing the reference sequence ==

  # Download a reference genome. Remove unnecessary comments from the fasta headers (>chr2 fbid="blalbal"; curator="hanswurst" => >chr2)
{{{
cat dmel-all-chromosome-r5.31|awk '{print $1}' > ../wg/dmel-5.31-short.fasta
}}}
  # Download the transposon sequences. Remove unnecessary comments from the fasta header
  # Filter TE elements that are to short (e.g.: < 40bp)
  # RepeatMask the reference genome using the transposon sequences as a custom library
{{{
perl RepeatMasker -no_is -nolow -norna --lib te-sequences.fasta -pa 4 reference-genome.fasta
}}}
  # Merge the repeat-masked reference genome and the TE sequences
{{{
cat reference-genome.fasta.masked te-sequences.fasta > combined-reference.fasta
}}}


== Obtaining a TE hierarcy ==

== Identify TE insertions ==

== Estimate population frequencies ==